src/Sylius/Behat/Context/Domain/ProductContext.php

```diff
@@ -74,6 +105,14 @@ public function iShouldBeNotifiedThatThisVariantIsInUseAndCannotBeDeleted()
     }
 
     /**
+     * @When I should be notified that this product is in use and cannot be deleted
+     */
+    public function iShouldBeNotifiedThatThisProductIsInUseAndCannotBeDeleted()
+    {
+        expect($this->sharedStorage->get('last_exception'))->toHaveType(DBALException::class);
```

Comment:
This method and this above have the same implementation, but they shouldn't. In first it's checked notification about **variant**, in second one about **product**. I think we should check also ``'last_exception'`` message, to be sure it was correct.